textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/GramBooleanQuery.java

```diff
@@ -399,11 +398,16 @@ public static GramBooleanQuery simplifyDNF(GramBooleanQuery query) {
 	 * The query tree generated by the translator is messy with possibly lots of redundant information.
 	 * This function transforms it into Disjunctive normal form (DNF), which is an OR of different ANDs.
 	 * 
-	 * For example, the original query of regex "abc?pqr?" is: 
-	 *  (((bpq AND abp) OR (abc AND cpq AND bcp)) AND (bpq OR cpq) AND (bpq OR cpq OR (pqr AND cpq) OR (pqr AND bpq)) AND (pqr OR bpq OR cpq) AND (abc OR abp))
-	 *  which is long and messey. After calling toDNF() and simplifyDNF(), the result query is:
-	 *  ((abp AND bpq) OR (abc AND bcp AND cpq))
-	 *  
+	 * To transform a tree to DNF form, the following laws are applied recursively from bottom to top:
+	 * Associative laws: (a OR b) OR c = a OR (b OR c) = a OR b OR c, when transforming OR nodes,
+	 * Distributive laws: a AND (b OR c) = (a AND b) OR (a AND c), when transforming AND nodes,
+	 * 
+	 * For each node, it's children will be transformed to DNF form first, then 
```

Comment:
I'll proofread more carefully in the future. I know I often make grammar errors.